src/plugins/intel_gpu/src/graph/common_utils/kernel_generator_base.hpp

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <cstddef>
+#include <type_traits>
+
+#include "intel_gpu/graph/kernel_impl_params.hpp"
+#include "intel_gpu/primitives/reorder.hpp"
+#include "intel_gpu/runtime/kernel_args.hpp"
+
+namespace micro {
+struct MicroKernelPackage;
+}  // namespace micro
+
+namespace ov::intel_gpu {
+
+using KernelParams = cldnn::kernel_arguments_desc;
+
+using KernelString = cldnn::kernel_string;
+using WorkGroupSizes = cldnn::work_group_sizes;
+using ScalarDescriptor = cldnn::scalar_desc;
+using Scalars = cldnn::scalars_desc;
+using ArgumentDescriptor = cldnn::argument_desc;
+using Arguments = cldnn::arguments_desc;
+using cldnn::WeightsReorderParams;
+using KernelLanguage = cldnn::kernel_language;
+
+struct KernelCode {
+    std::shared_ptr<KernelString> kernel_string;
+};
```

Comment:
Do we need this intermediate structure? `std::shared_ptr<KernelString> kernel_code` defined directly in KernelData looks a little bit simpler to use:
```
    kd.kernel_code = std::make_shared<KernelString>();
    kd.kernel_code->language = KernelLanguage::OCLC_V2;
    ...
```

Reply:
Agree. Removed this `KernelCode` structure